protocol/0035-LIQM-liquidity_monitoring.md

```diff
@@ -62,9 +62,29 @@ During the liquidity monitoring auction new or existing LPs can commit more stak
 ## What happens during the auction?
 
 The auction proceeds as usual. Please see the auction spec for details.
+## Frequency of checking for auction entry triggers
+
+ Through a sequence of actions which occur with the same timestamp the market may be moved into a state in which a liquidity auction is expected and then back out of said state. Ideally, liquidity auctions should only be entered when the market truly requires one as once entered a minimum auction length (controlled by `market.auction.minimumDuration`) must be observed. Even with a very short a minimum auction length, a market flickering between two states is suboptimal. 
+
+ To resolve this, the conditions for entering a liquidity auction should only be checked at the end of each batch of transactions occurring with an identical timestamp (in the current Tendermint implementation this is equivalent to once per block). At the end of each such period the auction conditions should be checked and the market moved into liquidity auction state if the conditions for entering a liquidity auction are satisfied. 
+The criteria for exiting any auction (liquidity or price monitoring) should be checked only on timestamp change (ie block boundary with Tendermint). This means that a market cannot leave a liquidity auction only to immediately re-enter it at the end of the block.
+
+ A liquidity provider amending LP provision order can reduce their stake (as long as total stake >= target stake) even if doing so would mean that at the end of block the system enters liquidity auction (because e.g. max open interest increased or because another LP was removed due to not meeting margin commitments). 
 
 ## Acceptance Criteria
 
 1. The scenarios in the feature test [0026-AUCT-auction_interaction.feature](https://github.com/vegaprotocol/vega/tree/develop/integration/features/verified/0026-AUCT-auction_interaction.feature) are verified and pass. (<a name="0035-LIQM-001" href="#0035-LIQM-001">0035-LIQM-001</a>)
 
-2. An incoming order that would consume `best_bid` or `best_offer` gets executed (unless it will also trigger price monitoring auction at the same time), the trades are generated and only then the market goes into a liquidity auction (because there is a peg missing to deploy the liquidity provision volume). (<a name="0035-LIQM-002" href="#0035-LIQM-002">0035-LIQM-002</a>)
\ No newline at end of file
+2. An incoming order that would consume `best_bid` or `best_offer` gets executed (unless it will also trigger price monitoring auction at the same time), the trades are generated. The market goes into a liquidity auction at the end of a block (because there is a peg missing to deploy the liquidity provision volume). (<a name="0035-LIQM-002" href="#0035-LIQM-002">0035-LIQM-002</a>)
```

Comment:
Seeing as we could be in a situation where these pegs are missing, I'm assuming any pegged order submissions during this time should be parked. Should we try and redeploy them as soon as we have the relevant pegs available again, or do we deploy all pegged orders at the end of the block/after liquidity auction?

I think it makes most sense to deploy them ASAP.

Another possible scenario to consider is how we handle LP submissions/amendments when these pegs are missing. Do we keep some LP orders on the book, and keep the LP submission as active, or remove all LP submissions that reference the missing peg instead, and try to redeploy it ASAP (like we would with pegged orders)? What about LP's that are pegged to the mid price? They could still stay deployed, in which case there's some implications WRT fees/equity like shares perhaps

Reply:
> Should we try and redeploy them as soon as we have the relevant pegs available again 

Yes. 

> remove all LP submissions that reference the missing peg instead, and try to redeploy it ASAP 

Ideally yes. 

> What about LP's that are pegged to the mid price? 

If they peg to mid price then they'll be removed. 

> in which case there's some implications WRT fees/equity like shares perhaps 

Hypothetically yes, but the incentive mechanism isn't exact science so I'd ignore. 

I'll update the spec if you think my answers are ok.